### PR TITLE
GODRIVER-523: Support new readConcern level "available"

### PR DESCRIPTION
**Pull request by igorKryvenko**

### CODE REVIEW

**Comment by skriptble** [comment]
Hi @igorKryvenko,

Thanks for the contribution! Would you mind updating the test files to also include the updates for linearizable and snapshot?

Thanks!
--Kris

**Comment by skriptble** [comment]
For reference the ticket is [GODIRVER-547](https://jira.mongodb.org/browse/GODRIVER-547).

**Comment by igorKryvenko** [comment]
Hi @skriptble.
Do you want me to add tests in this PR or create another one for it?

**Comment by skriptble** [comment]
Hi @igorKryvenko,
Adding it to this PR is fine, just add the ticket number in the commit.

--Kris

File: core/readconcern/readconcern.go [hunk 1]
```diff
@@ -40,6 +40,17 @@ func Linearizable() *ReadConcern {
...
 }
 
+// Available specifies that the query should return data from the instance with no guarantee
+// that the data has been written to a majority of the replica set members (i.e. may be rolled back).
+func Available() *ReadConcern {
+	return New(Level("available"))
+}
+
+// Only available for operations within multi-document transactions.
+func Snapshot() *ReadConcern {
```
Comment (skriptble) [inline core/readconcern/readconcern.go]: Getting a vet failure here because the comment doesn't start with "Snapshot ...".

Comment (igorKryvenko) [reply core/readconcern/readconcern.go]: Done.

**Pull request merged**